include/grpcpp/ext/otel_plugin.h

```diff
@@ -0,0 +1,99 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//
+
+#ifndef GRPCPP_EXT_OTEL_PLUGIN_H
+#define GRPCPP_EXT_OTEL_PLUGIN_H
+
+#include <grpc/support/port_platform.h>
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include <memory>
+
+#include "absl/functional/any_invocable.h"
+#include "absl/strings/string_view.h"
+#include "opentelemetry/metrics/meter_provider.h"
+
+namespace grpc {
+
+namespace internal {
+class OpenTelemetryPluginBuilderImpl;
+}  // namespace internal
+
+namespace experimental {
+// Metrics
+absl::string_view OpenTelemetryClientAttemptStartedInstrumentName();
+absl::string_view OpenTelemetryClientAttemptDurationInstrumentName();
+absl::string_view
+OpenTelemetryClientAttemptSentTotalCompressedMessageSizeInstrumentName();
+absl::string_view
+OpenTelemetryClientAttemptRcvdTotalCompressedMessageSizeInstrumentName();
+absl::string_view OpenTelemetryServerCallStartedInstrumentName();
+absl::string_view OpenTelemetryServerCallDurationInstrumentName();
+absl::string_view
+OpenTelemetryServerCallSentTotalCompressedMessageSizeInstrumentName();
+absl::string_view
+OpenTelemetryServerCallRcvdTotalCompressedMessageSizeInstrumentName();
+
+class OpenTelemetryPluginBuilder {
+ public:
+  OpenTelemetryPluginBuilder();
+  // If `SetMeterProvider()` is not called, no metrics are collected.
+  OpenTelemetryPluginBuilder& SetMeterProvider(
+      std::shared_ptr<opentelemetry::metrics::MeterProvider> meter_provider);
+  // If set, \a target_attribute_filter is called per channel to decide whether
+  // to record the target attribute on client or to replace it with "other".
+  // This helps reduce the cardinality on metrics in cases where many channels
+  // are created with different targets in the same binary (which might happen
+  // for example, if the channel target string uses IP addresses directly).
+  OpenTelemetryPluginBuilder& SetTargetAttributeFilter(
+      absl::AnyInvocable<bool(absl::string_view /*target*/) const>
+          target_attribute_filter);
+  // If set, \a generic_method_attribute_filter is called per call with a
+  // generic method type to decide whether to record the method name or to
+  // replace it with "other". Non-generic or pre-registered methods remain
+  // unaffected. If not set, by default, generic method names are replaced with
+  // "other" when recording metrics.
+  OpenTelemetryPluginBuilder& SetGenericMethodAttributeFilter(
+      absl::AnyInvocable<bool(absl::string_view /*generic_method*/) const>
+          generic_method_attribute_filter);
+  // Registers a global plugin that acts on all channels and servers running on
+  // the process.
+  // The most common way to use this API is -
```

Comment:
The rest of this comment should probably go above the class.

Reply:
done!